compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -900,4 +997,19 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         self.write_scalar(x.abs(), dest)?;
         interp_ok(())
     }
+
+    fn float_round_intrinsic<F>(
+        &mut self,
+        args: &[OpTy<'tcx, M::Provenance>],
+        dest: &PlaceTy<'tcx, M::Provenance>,
+        mode: rustc_apfloat::Round,
+    ) -> InterpResult<'tcx, ()>
+    where
+        F: rustc_apfloat::Float + rustc_apfloat::FloatConvert<F> + Into<Scalar<M::Provenance>>,
+    {
+        let x: F = self.read_scalar(&args[0])?.to_float()?;
+        let res = x.round_to_integral(mode).value;
+        self.write_scalar(res, dest)?;
```

Comment:
To fix the test failure, you'll have to do NaN adjustments here. Something like this:
```suggestion
        let res = self.adjust_nan(res, &[x]);
        self.write_scalar(res, dest)?;
```

Reply:
Thank you!! Fixed in 85056fd6465d99270f946b8740e2c57a2a462b5e.